### PR TITLE
Iris Tensorflow example

### LINKED ISSUE #184: Add iris model code to examples
### Description

The iris model code should show how to export the model, and should be linked from the README

### PR DESCRIPTION
**Pull request by 1vn**
Closes #184

---
Checklist:
- [ ] Run `make test` and `make lint`
- [ ] Test end to end manually (e.g. build/push all images, restart local operator, and run `cx refresh` in an example folder)
- [ ] Update documentation
- [ ] Update examples
- [ ] Alert team if dev environment changed
- [ ] Cherry-pick into release branches if it's a bugfix
- [ ] Delete the branch once it's merged

### CODE REVIEW

File: docs/apis/packaging-models.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 Zip the exported estimator output in your checkpoint directory, e.g.
 
 ```text
-$ ls export/estimator
+$ ls export/estimator/1560263597/
```
Comment (vishalbollu) [inline docs/apis/packaging-models.md]: What is the significance of this number? 1560263597

File: examples/models/tf.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# sources copied/modified from https://github.com/tensorflow/models/blob/master/samples/core/get_started/
```
Comment (vishalbollu) [inline examples/models/tf.py]: We should provide requirements.txt so that people can pip install and run this script to get a tf model

File: examples/models/tf.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import tensorflow as tf
+from sklearn.datasets import load_iris
+from sklearn.model_selection import train_test_split
+import shutil
+import os
+
+EXPORT_DIR = "iris_tf_export"
+
+
+def get_all_file_paths(directory):
```
Comment (vishalbollu) [inline examples/models/tf.py]: doesn't look like this is being used

File: pkg/workloads/tf_api/api.py [hunk 4]
```diff
@@ -379,6 +379,20 @@ def predict(deployment_name, api_name):
...
     return jsonify(response)
 
+def valid_model_dir(model_dir):
+    """
+    validates that model_dir has the expected directory tree, e.g:
+
+    {TF serving version number}/
+        saved_model.pb
+        variables/
+            variables.data-00000-of-00001
```
Comment (vishalbollu) [inline pkg/workloads/tf_api/api.py]: is it always going to be variables.data-00000-of-00001?

File: examples/models/tf.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+def input_fn(features, labels, batch_size, mode):
+    """An input function for training"""
+    dataset = tf.data.Dataset.from_tensor_slices((features, labels))
+    if mode == tf.estimator.ModeKeys.TRAIN:
+        dataset = dataset.shuffle(1000).repeat()
+    dataset = dataset.batch(batch_size)
+    dataset_it = dataset.make_one_shot_iterator()
+    images, labels = dataset_it.get_next()
+    return {"inputs": images}, labels
```
Comment (vishalbollu) [inline examples/models/tf.py]: Rename images, e.g. irises

File: docs/apis/packaging-models.md (hunk 1 above)
Comment (1vn) [reply docs/apis/packaging-models.md]: serving version or something

File: examples/models/tf.py (hunk 2 above)
Comment (1vn) [reply examples/models/tf.py]: what do you mean? this file has no deps

File: pkg/workloads/tf_api/api.py (hunk 4 above)
Comment (1vn) [reply pkg/workloads/tf_api/api.py]: not sure, which is why i didn't check for this

File: examples/models/tf.py (hunk 2 above)
Comment (vishalbollu) [reply examples/models/tf.py]: tensorflow and sklearn

File: pkg/workloads/tf_api/api.py (hunk 4 above)
Comment (vishalbollu) [reply pkg/workloads/tf_api/api.py]: gotcha thanks

File: examples/models/tf.py (hunk 2 above)
Comment (1vn) [reply examples/models/tf.py]: ah right...checked out

File: pkg/workloads/tf_api/api.py [hunk 6]
```diff
@@ -380,6 +380,21 @@ def predict(deployment_name, api_name):
...
+            variables.index
+    """
+    version = os.listdir(model_dir)[0]
```
Comment (deliahu) [inline pkg/workloads/tf_api/api.py]: Have you confirmed that it must be a digit? Is that a TF Serving thing?

File: pkg/workloads/tf_api/api.py [hunk 7]
```diff
@@ -380,6 +380,21 @@ def predict(deployment_name, api_name):
...
+    if not version.isdigit():
+        raise UserException("No versions of servable default found under base path in model_dir")
```
Comment (deliahu) [inline pkg/workloads/tf_api/api.py]: Can you also check for `saved_model.pb`? Or is it possible to have a servable model that doesn't have that?

Comment (deliahu) [inline pkg/workloads/tf_api/api.py]: Maybe append another sentence to the error message: "See docs.cortex.dev for how to properly package your TensorFlow model"`

File: pkg/workloads/tf_api/api.py (hunk 6 above)
Comment (1vn) [reply pkg/workloads/tf_api/api.py]: yes
